gsy_framework/utils.py

```diff
@@ -68,19 +68,26 @@ def convert_datetime_to_str_in_list(in_list: List, ui_format: bool = False):
 
 
 def generate_market_slot_list_from_config(sim_duration: duration, start_timestamp: DateTime,
-                                          slot_length: duration):
+                                          slot_length: duration, ignore_duration_check=False):
     """
-    Returns a list of all slot times in Datetime format
-    @param sim_duration: Total simulation duration
-    @param start_timestamp: Start datetime of the simulation
-    @param slot_length: Market slot length
-    @return: List with market slot datetimes
+    Returns a list of all slot times in Datetime format.
+
+    Args:
+        sim_duration: Total simulation duration
+        start_timestamp: Start datetime of the simulation
+        slot_length: Market slot length
+        ignore_duration_check: Ignores the check if each timestamp is in the simulation duration.
+                               Useful for calling the method from context where no config has been
+                               set
```

Comment:
Sorry, I still don't get why the lack of a config is the reason for this. AFAIS, the slot list is only created from imput variables. But probably I oversaw it. Can you help me please?
(BTW; also a reason fo renaming this method and omit the `_from_config` suffix IMO).

Reply:
Unfortunately the duration check needs either GlobalConfig, or a constructed SimulationConfig object (in the context of this function, GlobalConfig is used). Therefore, if the GlobalConfig is not set correctly (which is not possible in gsy-web) then such a parameter is needed to circumvent the duration check.

Reply:
🤦  darn, overlooked this. Indeed.